src/control-plane/backend/lambda/api/service/segment.ts

```diff
@@ -0,0 +1,337 @@
+    try {
+      const { appId } = req.query;
+      const { segmentId } = req.params;
+      const segment = await segmentStore.get(appId as string, segmentId as string);
+
+      return res.status(200).json(new ApiSuccess(segment));
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  public async update(req: Request, res: Response, next: NextFunction) {
+    try {
+      const {
+        segmentId,
+        name,
+        description,
+        appId,
+        refreshSchedule,
+        criteria,
+        eventBridgeRuleArn,
+      } = req.body;
+      const segmentDdbItem = await segmentStore.get(appId as string, segmentId as string);
+      if (segmentDdbItem === undefined) {
+        return res.status(400).send(new ApiFail(`Segment with id ${segmentId} is not found`));
+      }
+
+      const updatedItem = {
+        ...segmentDdbItem,
+        name,
+        description,
+        lastUpdateBy: res.get('X-Click-Stream-Operator') ?? 'Unknown operator',
+        lastUpdateAt: Date.now(),
+        refreshSchedule,
+        criteria,
+        eventBridgeRuleArn,
+      };
+
+      // Update EventBridge rule
+      if (refreshSchedule.cron !== 'Manual' && !!refreshSchedule.cronExpression) {
+        const rule = await this.putEventBridgeRule(updatedItem as Segment);
+        updatedItem.eventBridgeRuleArn = rule.RuleArn;
+      }
+      // Clean up EventBridge rule when cron type is set to Manual
+      if (refreshSchedule.cron === 'Manual' && segmentDdbItem.eventBridgeRuleArn !== '') {
+        await this.deleteEventBridgeRule(segmentId);
+        updatedItem.eventBridgeRuleArn = '';
+      }
+
+      // Save segment setting to DDB
+      const id = await segmentStore.update(updatedItem as SegmentDdbItem);
+      logger.info('Update segment setting id: ' + id);
+
+      return res.status(201).json(new ApiSuccess({ id }, 'Segment updated successfully.'));
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  public async delete(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { segmentId } = req.params;
+      const { appId } = req.query;
+      const segment = await segmentStore.get(appId as string, segmentId as string);
+      if (!segment) {
+        return res.status(400).send(new ApiFail(`Segment with id ${segmentId} is not found`));
+      }
+
+      // Clean up EventBridge rule
+      if (segment.eventBridgeRuleArn !== '') {
+        await this.deleteEventBridgeRule(segmentId);
+      }
+
+      await segmentStore.delete(appId as string, segmentId as string);
+
+      return res.status(200).send(new ApiSuccess({ segmentId }, 'Segment deleted successfully.'));
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  public async listJobs(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { segmentId } = req.params;
+      const segmentJobs = await segmentStore.listJobs(segmentId as string, SEGMENT_JOBS_LIST_LIMIT);
```

Comment:
Do we support pagination?

Reply:
For segment jobs, we'll get latest 30 (by default) segment jobs. No pagination is needed.